librz/util/print.c

```diff
@@ -1726,43 +1824,31 @@ RZ_API int rz_print_get_cursor(RzPrint *p) {
 	return p->cur_enabled ? p->cur : 0;
 }
 
-RZ_API int rz_print_jsondump(RzPrint *p, const ut8 *buf, int len, int wordsize) {
-	ut16 *buf16 = (ut16 *)buf;
-	ut32 *buf32 = (ut32 *)buf;
-	ut64 *buf64 = (ut64 *)buf;
-	// TODDO: support p==NULL too
-	if (!p || !buf || len < 1 || wordsize < 1) {
-		return 0;
-	}
+/**
+ * \brief Print dump in json format
+ * \param p RzPrint instance
+ * \param buf Buffer to print to
+ * \param len Print only this many bytes
+ * \param wordsize Size of a word in bits
+ * \return Dump JSON string
+ */
+RZ_API RZ_OWN char *rz_print_jsondump_str(RZ_NONNULL RzPrint *p, RZ_NONNULL const ut8 *buf, int len, int wordsize) {
```

Comment:
This function could be probably removed from `librz/util/` and put directly in the `px` handler code, which should use `pj` allocated by the `RzCmdStateOutput`